pkg/metrics/metrics.go

```diff
@@ -199,12 +200,37 @@ var (
 		},
 	)
 
-	// subscriptionSyncCounters keeps a record of the Prometheus counters emitted by
-	// Subscription objects. The key of a record is the Subscription name, while the value
-	//  is struct containing label values used in the counter
-	subscriptionSyncCounters = make(map[string]subscriptionSyncLabelValues)
+	subscriptionSyncCounters = newSubscriptionSyncCounter()
 )
 
+// subscriptionSyncCounter keeps a record of the Prometheus counters emitted by
+// Subscription objects. The key of a record is the Subscription name, while the value
+// is struct containing label values used in the counter. Read and Write access are
+// protected by mutex.
+type subscriptionSyncCounter struct {
+	counters     map[string]subscriptionSyncLabelValues
+	countersLock sync.RWMutex
+}
+
+func newSubscriptionSyncCounter() subscriptionSyncCounter {
+	return subscriptionSyncCounter{
+		counters: make(map[string]subscriptionSyncLabelValues),
+	}
+}
+
+func (s *subscriptionSyncCounter) setValues(key string, val subscriptionSyncLabelValues) {
+	s.countersLock.Lock()
+	defer s.countersLock.Unlock()
+	s.counters[key] = val
+}
+
+func (s *subscriptionSyncCounter) readValues(key string) (subscriptionSyncLabelValues, bool) {
+	s.countersLock.RLock()
+	defer s.countersLock.RUnlock()
+	val, ok := s.counters[key]
+	return val, ok
```

Comment:
```suggestion
	return s.counters[key]
```

Reply:
Wow, I would have thought that this behaved the same way as it does for [functions](https://go.dev/play/p/J6HLJ0ZZQH_6), but for some reason [the bool isn't returned](https://go.dev/play/p/JSvaVQCXiKX) doesn't work when trying to use the result of a map lookup to satisfy

Reply:
I suspect this is because you don't have to receive the value as shown [here](https://go.dev/play/p/gXjkM1_zVPs).

Reply:
Courtesy of @benluddy:

[From the Go Docs](https://go.dev/ref/spec#Index_expressions):
> An index expression on a map a of type map[K]V used in an [assignment statement](https://go.dev/ref/spec#Assignment_statements) or initialization of the special form
> v, ok = a[x]
> v, ok := a[x]
> var v, ok = a[x]
> yields an additional untyped boolean value.